src/comments.go

```diff
@@ -17,72 +15,52 @@ type GithubClient struct {
 	Client *github.Client
 }
 
-func DeclareClient(client *GithubClient) {
-	token := os.Getenv("GITHUB_TOKEN")
+// NewClient creates a new GithubClient with the provided token
+func NewClient(token string, client *GithubClient) error {
 	if token == "" {
-		log.Fatal("Unauthorized: No token present")
+		return fmt.Errorf("unauthorized: no token present")
 	}
 	client.Ctx = context.Background()
 	client.Client = github.NewClient(nil).WithAuthToken(token)
+	return nil
 }
 
-func GetUserComments(pull int, org, repo, content, filePath string) {
-	var client GithubClient
-	DeclareClient(&client)
-
+func (client *GithubClient) GetUserComments(pull int, org, repo, content, filePath string) (int64, error) {
 	comments, _, err := client.Client.Issues.ListComments(client.Ctx, org, repo, pull, nil)
 	if err != nil {
-		log.Fatalf("Error: %v\n", err)
-	}
-	commentBody := content
-	if filePath != "" {
-		commentBody = files.ParseFileContent(filePath)
+		return 0, fmt.Errorf("error listing comments: %v", err)
 	}
+
+	commentBody := files.GetCommentBody(content, filePath)
 	for _, comment := range comments {
 		if strings.Contains(*comment.Body, commentBody) {
-			fmt.Println(*comment.ID)
-			return
+			return *comment.ID, nil
 		}
 	}
-	log.Error("Comment not found!")
-}
 
-func CreateComment(pull int, org, repo, content, filePath string) {
-	var client GithubClient
-	DeclareClient(&client)
+	return 0, fmt.Errorf("comment not found")
```

Comment:
Same here, try to include more info so that our DebugEx is improved.

```suggestion
	return 0, fmt.Errorf("Comment not found")
```

Reply:
fixed